### DISCUSSION TITLE
How to host the 'wpf control' in uno?

### DISCUSSION THREAD

**Discussion by SongOfYouth** [unoplatform/uno#11174] [Q&A] [answered]
## What would you like to be added:
someway to host a wpf/winform control even only work on windows.

## Why is this needed:
we has a heavy third-part control implemented with winform/wpf, and if we try to uno, it's the greatest challenge. if it's possible,we can port it to uno at the first step, then we have the time to really fix it!

**Comment by jeromelaban** [comment]
Thanks for the request. Can you explain a bit more your scenario? Is it that you want a WinAppSDK app to use WPF controls, or the other way around?

**Reply by SongOfYouth** [reply]
thanks for the reply.
Yes,I wonder if it's possible that to use a 3-part winform control in winui, and work well with uno? before we find the replacement, we need it in our solution.

**Reply by jeromelaban** [reply, marked as answer]
Thanks. Hosting is currently available the other way around ([Microsoft's Xaml Islands](https://learn.microsoft.com/en-us/windows/apps/desktop/modernize/host-standard-control-with-xaml-islands) and [Uno Islands](https://platform.uno/docs/articles/guides/uno-islands.html)). Note that Uno Islands is only available on Windows at this time.

Integrating WinForms or WPF controls in a WinAppSDK or Uno app (using GTK or WPF backends) is currently not supported, mainly because of compositing concerns. It involves synchronizing WinAppSDK's and Uno's rendering to what WinForms or WPF are rendering.

**Reply by SongOfYouth** [reply]
some days,I have tended to use the replacement which implemented on web, but we can only host the web via webview not webview2 on uno platform, the aproch of webview is not recommanded now.
